Source/AwsCommonRuntimeKit/crt/CRTError.swift

```diff
@@ -1,18 +1,30 @@
-//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-//  SPDX-License-Identifier: Apache-2.0.
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
 import AwsCCommon
 
-public struct CRTError: Error {
+extension CRTError {
+    public init(fromErrorCode errorCode: Int32) {
+        self = CRTError(rawValue: errorCode) ?? .UNKNOWN_ERROR_CODE
```

Comment:
I'm confused about the `??`

I don't see how `CRTError(rawValue: 0)` could evaluate to `nil`? But I don't know Swift super well yet...

Should this be like
`self.rawValue = errorCode !=0 ? errorCode : UNKNOWN_ERROR_CODE` ???

Reply:
The nil-coalescing operator (??) is used to supply a default value in case the Optional instance is nil. It's shortform of something like 
```
if CRTError(rawValue: errorCode) != nil { 
    self = CRTError(rawValue:errorCode) 
}
else {
    self = .UNKNOWN_ERROR_CODE
}
``` 
So if we pass any integer value that is not defined in CRTError like -100, it will evaluate it to UNKNOWN_ERROR_CODE instead of None, which matches the C implementation.

Reply:
The CRTError (rawValue: errorCode) converts rawValue to an Enum Case so if we pass 1, it will convert it to CRTError.AWS_ERROR_OOM. This is a handy function to convert the integer value of aws_last_error() to a matching enum case.